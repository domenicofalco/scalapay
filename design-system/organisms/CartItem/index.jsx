import styled from "styled-components";
import { base } from "fonts";
import { useEffect, useState } from "react";

const Wrapper = styled.div``;

const Text = styled.p`
  font-size: ${base};
`;

const Button = styled.button`
  border: 1px solid #000;
  width: 20px;
  margin: 0 4px;
  padding: 3px;
  cursor: pointer;
  background: none;
`;

export default function CartItem({
  price: { amount, currency },
  brand,
  category,
  name,
  qt,
  id,
  updateUnitQuantity = () => null
}) {
  const [quantity, setQuantity] = useState(qt);
  const add = () => {
    setQuantity(quantity + 1);
  };

  const remove = () => {
    if (quantity !== 0) {
      setQuantity(quantity - 1);
    }
  };

  useEffect(() => {
    updateUnitQuantity(id, quantity);
  }, [quantity]);

  return (
    <Wrapper>
      <Text>
        {brand} - {name} ({category})
      </Text>
      <Text>
        quantity {quantity}
        <Button onClick={add}>+</Button>
        <Button onClick={remove}>-</Button>
      </Text>
      <Text>
        Price per unit: {amount} {currency}
      </Text>
      <Text>SKU {id}</Text>
    </Wrapper>
  );
}
