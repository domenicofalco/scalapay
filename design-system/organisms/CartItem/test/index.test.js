import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import CartItem from "../";

const props = {
  price: { amount: "800", currency: "EUR" },
  brand: "apple",
  category: "mobile device",
  name: "iphone 13",
  qt: 1,
  id: "00000",
  updateUnitQuantity: () => null
};

afterEach(cleanup);

describe("organisms => CartItem", () => {
  it("It should match snapshots", () => {
    const { asFragment } = render(<CartItem {...props}>CartItem</CartItem>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("It should increase qt on + click", () => {
    const { container, getByText } = render(
      <CartItem {...props}>CartItem</CartItem>
    );
    const addButton = container.querySelector(".add-qt");
    const qtContainer = getByText(/Qt/i);

    expect(qtContainer.textContent).toBe("Qt 1");
    fireEvent.click(addButton);
    expect(qtContainer.textContent).toBe("Qt 2");
  });

  it("It should decrease qt on - click", () => {
    const { container, getByText } = render(
      <CartItem {...props}>CartItem</CartItem>
    );
    const removeButton = container.querySelector(".remove-qt");
    const qtContainer = getByText(/Qt/i);

    expect(qtContainer.textContent).toBe("Qt 1");
    fireEvent.click(removeButton);
    expect(qtContainer.textContent).toBe("Qt 0");
  });
});
