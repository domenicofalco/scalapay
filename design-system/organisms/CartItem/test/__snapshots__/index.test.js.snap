// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms => CartItem It should match snapshots 1`] = `
<DocumentFragment>
  <div>
    <p
      class="Paragraph-sc-14i1zm6-0 CartItem__Text-sc-cto248-1 lbtMbQ iMMURT"
    >
      apple - iphone 13 (mobile device)
    </p>
    <p
      class="Paragraph-sc-14i1zm6-0 CartItem__Text-sc-cto248-1 lbtMbQ iMMURT"
    >
      Price per unit: 
      <b>
        800 EUR
      </b>
    </p>
    <div
      class="CartItem__Quantity-sc-cto248-0 fzcZsj"
    >
      <p
        class="Paragraph-sc-14i1zm6-0 CartItem__Text-sc-cto248-1 lbtMbQ iMMURT"
      >
        Qt 1
      </p>
      <button
        class="Button__roundIcon-sc-6egjvv-2 feZvzI add-qt"
        data-variant="roundIcon"
        style="margin-right: 5px;"
        type="button"
      >
        +
      </button>
      <button
        class="Button__roundIcon-sc-6egjvv-2 feZvzI remove-qt"
        data-variant="roundIcon"
        style="margin-right: 5px;"
        type="button"
      >
        -
      </button>
    </div>
  </div>
</DocumentFragment>
`;
